builtin/logical/mysql/path_role_create.go

```diff
@@ -51,9 +52,15 @@ func (b *backend) pathRoleCreateRead(
 	}
 
 	// Generate our username and password. MySQL limits user to 16 characters
-	displayName := req.DisplayName
-	if len(displayName) > 10 {
-		displayName = displayName[:10]
+	displayName := name
+	ul, ok := data.GetOk("username_length")
+	if ok == true {
+		usernameLength = ul.(int)
+	} else {
+		usernameLength = 10
+	}
+	if len(displayName) > usernameLength {
```

Comment:
`role.UsernameLength` here is confusing because it is used to restrict the length of display name alone as opposed to restrict the length of the *final* username. We should instead do that. Currently, the `username` is capped at the length of 16 (which I think should be bigger anyways).